### ISSUE TITLE
Shader Validation Skipped

### ISSUE THREAD

**Issue by Binero** [gfx-rs/naga#176]
The following shaders are not validated:

```glsl
#version 450

layout(location = 0) in vec2 a_position;
layout(location = 0) out vec2 out_tile_index;

void main() {
  const vec2 position = a_position * 0.8;

  gl_Position = vec4(position.x, position.y, 0.8, 1.0);
  out_tile_index = (a_position + 1.0) / 2.0 * 2.0;
}
```

```glsl
#version 450

layout(location = 0) in vec2 in_tile_index;
layout(location = 0) out vec4 f_colour;

layout(set = 0, binding = 0) buffer readonly Map { uint a_test[]; };

void main() {
  uint tile_index = a_test[uint(in_tile_index.y) * 2 + uint(in_tile_index.x)];

  const vec3 colours[] = vec3[](vec3(1.0, 0.0, 0.0), vec3(0.0, 1.0, 0.0),
                                vec3(0.0, 0.0, 1.0), vec3(1.0, 1.0, 0.0));
  vec3 colour =
      tile_index < colours.length() ? colours[tile_index] : vec3(0.0, 0.0, 0.0);

  f_colour = vec4(colour, 1.0);
}
```

```
[0.040029 WARN]()(no module): Unable to find layer: VK_LAYER_KHRONOS_validation
[0.055584 WARN]()(wgpu_core::device): Failed to parse shader SPIR-V code: UnsupportedBuiltIn(4)
[0.055656 WARN]()(wgpu_core::device): Shader module will not be validated
[0.055858 WARN]()(no module): Unknown decoration NonWritable
[0.056285 WARN]()(wgpu_core::device): Failed to parse shader SPIR-V code: UnsupportedInstruction(Function, ConvertFToU)
[0.056316 WARN]()(wgpu_core::device): Shader module will not be validated
```

**Comment by kvark** [comment]
@Binero just to clarify, you were running on wgpu-0.6, right? So these validation errors would be for naga-0.2?

**Comment by kvark** [comment]
`NonWritable` support landed in #182, everything else was already fixed in `master`. It just needs to make its way into `wgpu` master.
